admin/test/test_release.py

```diff
@@ -661,3 +667,637 @@ def test_publish_non_release_fails(self):
+        S3 which correspond to packages including those already on S3.
+        """
+        cli_package = 'clusterhq-flocker-cli-0.3.3-0.dev.7.noarch.rpm'
+        existing_s3_keys = {
+            os.path.join(self.target_key, cli_package): 'old-cli-package',
+        }
+
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: {},
+                self.alternative_bucket: existing_s3_keys.copy(),
+            },
+        )
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files={}),
+            packages=self.packages,
+        )
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.alternative_package_directory,
+            target_bucket=self.alternative_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files={}),
+            packages=self.packages,
+        )
+
+        index_path = os.path.join(self.target_key, 'repodata', 'repomd.xml')
+        self.assertNotEqual(
+            aws.s3_buckets[self.target_bucket][index_path],
+            aws.s3_buckets[self.alternative_bucket][index_path])
+
+    def test_create_repository_accounts_for_new_packages(self):
+        """
+        Calling :func:`update_repo` uploads new repository metadata files to
+        S3 which correspond to packages including those in the source
+        repository but not already on S3.
+        """
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: {},
+                self.alternative_bucket: {},
+            },
+        )
+
+        repo_contents = {
+            'clusterhq-flocker-cli-0.3.3-0.dev.7.noarch.rpm': 'cli-package',
+            'clusterhq-flocker-node-0.3.3-0.dev.7.noarch.rpm': 'node-package',
+        }
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files=repo_contents),
+            packages=[],
+        )
+
+        self.update_repo(
+            aws=aws,
+            yum=FakeYum(),
+            rpm_directory=self.alternative_package_directory,
+            target_bucket=self.alternative_bucket,
+            target_key=self.target_key,
+            source_repo=self.create_fake_repository(files=repo_contents),
+            packages=self.packages,
+        )
+
+        index_path = os.path.join(self.target_key, 'repodata', 'repomd.xml')
+        self.assertNotEqual(
+            aws.s3_buckets[self.target_bucket][index_path],
+            aws.s3_buckets[self.alternative_bucket][index_path])
```

Comment:
Or why we care that the index of `target_bucket` is different that `alternative_bucket`.